src/Symfony/Component/Filesystem/Path.php

```diff
@@ -0,0 +1,990 @@
+     *                               multi-byte character handling in extension).
+     *
+     * @return string The extension of the file path (without leading dot).
+     *
+     * @since 1.1 Added method.
+     * @since 2.0 Method now fails if $path is not a string.
+     */
+    public static function getExtension(string $path, bool $forceLowerCase = false): string
+    {
+        if ('' === $path) {
+            return '';
+        }
+
+        $extension = pathinfo($path, PATHINFO_EXTENSION);
+
+        if ($forceLowerCase) {
+            $extension = self::toLower($extension);
+        }
+
+        return $extension;
+    }
+
+    /**
+     * Returns whether the path has an extension.
+     *
+     * @param string            $path       The path string.
+     * @param string|string[]|null $extensions If null or not provided, checks if
+     *                                      an extension exists, otherwise
+     *                                      checks for the specified extension
+     *                                      or array of extensions (with or
+     *                                      without leading dot).
+     * @param bool              $ignoreCase Whether to ignore case-sensitivity
+     *                                      (requires mbstring extension for
+     *                                      correct multi-byte character
+     *                                      handling in the extension).
+     *
+     * @return bool Returns `true` if the path has an (or the specified)
+     *              extension and `false` otherwise.
+     *
+     * @since 1.1 Added method.
+     * @since 2.0 Method now fails if $path or $extensions have invalid types.
+     */
+    public static function hasExtension(string $path, $extensions = null, bool $ignoreCase = false): bool
+    {
+        if ('' === $path) {
+            return false;
+        }
+
+        $actualExtension = self::getExtension($path, $ignoreCase);
+
+        // Only check if path has any extension
+        if (empty($extensions)) {
+            return '' !== $actualExtension;
+        }
+
+        foreach ($extensions as $key => $extension) {
+            if ($ignoreCase) {
+                $extension = self::toLower($extension);
+            }
+
+            // remove leading '.' in extensions array
+            $extensions[$key] = ltrim($extension, '.');
+        }
+
+        return in_array($actualExtension, $extensions, true);
+    }
+
+    /**
+     * Changes the extension of a path string.
+     *
+     * @param string $path      The path string with filename.ext to change.
+     * @param string $extension New extension (with or without leading dot).
+     *
+     * @return string The path string with new file extension.
+     *
+     * @since 1.1 Added method.
+     * @since 2.0 Method now fails if $path or $extension is not a string.
+     */
+    public static function changeExtension(string $path, string $extension): string
```

Comment:
https://3v4l.org/qnrU3

is this expected? In this case, my extension is `.html.twig` and I would expect it to change to `.xml` not `.html.xml`

Reply:
And what about `my.example.html`? There's no one implementation that satisfies all assumptions.

This uses PHP's `PATHINFO_EXTENSION` under the hood, hence this returns what PHP developers should expect.

Reply:
`withExtension`?